set_user.c

```diff
@@ -98,65 +43,28 @@
 #include "utils/guc.h"
 #include "utils/memutils.h"
 #include "utils/syscache.h"
-#ifdef HAS_VARLENA_H
-#include "utils/varlena.h"
-#endif /* HAS_VARLENA_H */
 
 #include "set_user.h"
-#define WHITELIST_WILDCARD	"*"
-#define SUPERUSER_AUDIT_TAG	"AUDIT"
 
 PG_MODULE_MAGIC;
 
+#include "compatibility.h"
+
+#define WHITELIST_WILDCARD	"*"
+#define SUPERUSER_AUDIT_TAG	"AUDIT"
+
 static char *save_log_statement = NULL;
 static Oid save_OldUserId = InvalidOid;
 static char *reset_token = NULL;
 static ProcessUtility_hook_type prev_hook = NULL;
 
-#ifdef HAS_ALTER_SYSTEM
-/* 9.4 & up */
 static bool Block_AS = false;
-#endif
-
-#ifdef HAS_COPY_PROGRAM
-/* 9.3 & up */
-static bool Block_CP = false;
-#endif
-
 static bool Block_LS = false;
+static bool Block_CP = false;
```

Comment:
you can reduce your diff if you place Block_CP above Block_LS

Reply:
good eye